xl/trax/track.py

```diff
@@ -660,12 +668,12 @@ def get_tag_search(self, tag, format=True, artist_compilations=False,
         """
         extraformat = ""
         if tag == "albumartist":
-            if artist_compilations and self.__tags.get('__compilation'):
-                value = self.__tags.get('albumartist', None)
-                tag = 'albumartist'
-                extraformat += " ! __compilation==__null__"
+            if self.__tags.get('albumartistsort'):
+                value = self.__tags.get('albumartistsort')
+            elif artist_compilations and self.__tags.get('__compilation'):
+                value = self.__tags.get('albumartist', _VARIOUSARTISTSSTR)
             else:
-                value = self.__tags.get('artist')
+                value = self.__tags.get('albumartist', _UNKNOWNSTR)
```

Comment:
So if you're asking to search by albumartist, and there is no tag, it uses unknown instead of using the artist?